tests/test_datasets/test_dataset_functions.py

```diff
@@ -379,3 +380,112 @@ def test_create_dataset_numpy(self):
             paper_url=paper_url
         )
         dataset.publish()
+
+    def test_create_dataset_pandas(self):
+        # pandas is only a optional dependency and we need to skip the test if
+        # it is not installed.
+        pd = pytest.importorskip('pandas')
+        data = [
+            ['a', 'sunny', 85.0, 85.0, 'FALSE', 'no'],
+            ['b', 'sunny', 80.0, 90.0, 'TRUE', 'no'],
+            ['c', 'overcast', 83.0, 86.0, 'FALSE', 'yes'],
+            ['d', 'rainy', 70.0, 96.0, 'FALSE', 'yes'],
+            ['e', 'rainy', 68.0, 80.0, 'FALSE', 'yes'],
+            ['f', 'rainy', 65.0, 70.0, 'TRUE', 'no'],
+            ['g', 'overcast', 64.0, 65.0, 'TRUE', 'yes'],
+            ['h', 'sunny', 72.0, 95.0, 'FALSE', 'no'],
+            ['i', 'sunny', 69.0, 70.0, 'FALSE', 'yes'],
+            ['j', 'rainy', 75.0, 80.0, 'FALSE', 'yes'],
+            ['k', 'sunny', 75.0, 70.0, 'TRUE', 'yes'],
+            ['l', 'overcast', 72.0, 90.0, 'TRUE', 'yes'],
+            ['m', 'overcast', 81.0, 75.0, 'FALSE', 'yes'],
+            ['n', 'rainy', 71.0, 91.0, 'TRUE', 'no']
+        ]
+        column_names = ['rnd_str', 'outlook', 'temperature', 'humidity',
+                        'windy', 'play']
+        df = pd.DataFrame(data, columns=column_names)
+        # enforce the type of each column
+        df['outlook'] = df['outlook'].astype('category')
+        df['windy'] = df['windy'].astype('category')
+        df['play'] = df['play'].astype('category')
+        # meta-information
+        name = 'Pandas_testing_dataset'
+        description = 'Synthetic dataset created from a Pandas DataFrame'
+        creator = 'OpenML tester'
+        collection_date = '01-01-2018'
+        language = 'English'
+        licence = 'MIT'
+        default_target_attribute = 'play'
+        citation = 'None'
+        original_data_url = 'http://openml.github.io/openml-python'
+        paper_url = 'http://openml.github.io/openml-python'
+        dataset = openml.datasets.functions.create_dataset(
+            name=name,
+            description=description,
+            creator=creator,
+            contributor=None,
+            collection_date=collection_date,
+            language=language,
+            licence=licence,
+            default_target_attribute=default_target_attribute,
+            row_id_attribute=None,
+            ignore_attribute=None,
+            citation=citation,
+            attributes='auto',
+            data=df,
+            format='arff',
+            version_label='test',
+            original_data_url=original_data_url,
+            paper_url=paper_url
+        )
+        dataset.publish()
```

Comment:
Could you please download and check that everything was converted and uploaded properly?

Reply:
I assumed that I can do that with something like:

```python
upload_id = dataset.publish()
dataset_uploaded = openml.datasets.get_dataset(upload_id)
```

However, I receive the following exception:

```pytb
openml.exceptions.OpenMLServerException: https://test.openml.org/api/v1/xml/data/features/5915 returned code 273: Dataset not processed yet
```

What is the usual way that you are doing it.

Reply:
One way could be something like that:

```python
        upload_id = dataset.publish()
        while True:
            try:
                dataset_uploaded = openml.datasets.get_dataset(upload_id)
            except OpenMLServerException as msg_exception:
                if "code 273" in str(msg_exception):
                    pass
                else:
                    raise msg_exception
```

but I am not sure how long it can take for the dataset to be processed and we could get in some timeout nightmare.

Reply:
@glemaitre We already had this problem in #547. We decided to only compare based on the arff and we also implemented 2 functions which get the arff and format based on the id (those are available). I have to refactor my pr a bit more, but it should be available tomorrow.

Reply:
So do you mean to use this function:

https://github.com/openml/openml-python/pull/547/files#diff-2ffad3a1ae43bbcc0e841e2f1df8967bR692

So I would probably wait that your PR get first in and I will manage the conflict then.

Reply:
yes, @mfeurer are you okay with this?

Reply:
This looks good at first glance.